src/main/proto/firrtl.proto

```diff
@@ -0,0 +1,451 @@
+syntax = "proto3";
+
+package hardware.chips.chisel3;
+
+option java_package = "hardware.chips.chisel3.firrtl";
+
+message Firrtl {
+  repeated Circuit circuit = 1;
+
+  message SourceInfo {
+    message None {
+      enum Reason {
+        NONE_SOURCE_INFO_REASON_UNKNOWN = 0;
+        NONE_SOURCE_INFO_REASON_UNLOCATABLE = 1;
+        NONE_SOURCE_INFO_REASON_SUPPRESSED = 2;
+        NONE_SOURCE_INFO_REASON_DEPRECATED = 3;
+      }
+
+      // Required.
+      Reason reason = 1;
+    }
+
+    message Position {
+      // Required.
+      string filename = 1;
+      // Required.
+      uint32 line = 2;
+      // Required.
+      uint32 column = 3;
+    }
+
+    // Required.
+    oneof source_info {
+      None none = 1;
+      Position position = 2;
+    }
+  }
+
+  message Top {
+    // Required.
+    string name = 1;
+  }
+
+  message Circuit {
+    repeated Module module = 1;
+    repeated Top top = 2;
+  }
+
+  message Module {
```

Comment:
If we want to match the scala ir, then Module -> DefModule, and UserModule -> Module.

Reply:
Oh, it stands for "Definition", but for better or for worse, its the name of the corresponding firrtl ir node (in firrtl/ir/IR.scala).